modules/core/keeper/msg_server.go

```diff
@@ -69,17 +69,23 @@ func (k Keeper) UpdateClient(goCtx context.Context, msg *clienttypes.MsgUpdateCl
 func (k Keeper) UpgradeClient(goCtx context.Context, msg *clienttypes.MsgUpgradeClient) (*clienttypes.MsgUpgradeClientResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	upgradedClient, err := clienttypes.UnpackClientState(msg.ClientState)
+	upgradedClientState, err := k.cdc.Marshal(msg.ClientState)
 	if err != nil {
 		return nil, err
 	}
-	upgradedConsState, err := clienttypes.UnpackConsensusState(msg.ConsensusState)
+
+	upgradedConsensusState, err := k.cdc.Marshal(msg.ConsensusState)
```

Comment:
```suggestion
        // Backwards Compatibility: the light client module is expecting
        // to unmarshal an interface so we marshal the consensus state Any
	upgradedConsensusState, err := k.cdc.Marshal(msg.ConsensusState)
```

some sort of in line code doc like this can be useful